man/systemd.slice.xml

```diff
@@ -104,9 +104,31 @@
     <para>Slice unit files may include [Unit] and [Install] sections, which are described in
     <citerefentry><refentrytitle>systemd.unit</refentrytitle><manvolnum>5</manvolnum></citerefentry>.</para>
 
-    <para>Slice files may include a [Slice] section. Options that may be used in this section are shared with other unit types.
-    These options are documented in
+    <para>Slice files may include a [Slice] section. Many options that may be used in this section are shared
+    with other unit types.  These options are documented in
     <citerefentry><refentrytitle>systemd.resource-control</refentrytitle><manvolnum>5</manvolnum></citerefentry>.</para>
+
+    <para>The options specific to the [Slice] section of slice units are the following:</para>
+
+    <variablelist class='unit-directives'>
+      <varlistentry>
+        <term><varname>ConcurrencyHardMax=</varname></term>
+        <term><varname>ConcurrencySoftMax=</varname></term>
+
+        <listitem><para>Configures a hard and a soft limit on the maximum number of units assigned to this
+        slice (or any descendent slices) that may be active at the same time. If the hard limit is reached no
+        further units associated with the slice may be activated, and their activation will fail with an
+        error. If the soft limit is reached any further requested activation of units will be queued, but no
+        immediate error is generated. The queued activation job will remain queued until the number of
+        concurrent active units within the slice is below the limit again.</para>
```

Comment:
Units activated under one slice can be generally in any order (like e.g. services started in parallel in `systemd.slice`). Concurrency limit will make this randomness more visible.
It's OK with queueing but rejecting jobs may be surprising. Add a note about this non-deterministic nature?

Reply:
That's not quite true. The order is not random. the runq is a prioq and jobs are dispatched in the order defined by unit_compare_priority(). Which looks at unit type, nice level, cpu weight, and finally at the unit name.

but true, we might want to document that here, at least vaguely (only vaguely, not precisely, since I think we hould have some freedom to change things later.

Reply:
i extended the docs on this